tools/Microsoft.Build.Locator/Microsoft.Build.Locator.csproj

```diff
@@ -0,0 +1,67 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{158E7760-6F87-446C-B782-18E23BF54DA6}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Microsoft.Build.Locator</RootNamespace>
+    <AssemblyName>Microsoft.Build.Locator</AssemblyName>
+    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
+    <DefineConstants>TRACE;DEBUG;FEATURE_VISUALSTUDIOSETUP</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE;FEATURE_VISUALSTUDIOSETUP</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.VisualStudio.Setup.Configuration.Interop, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
```

Comment:
I'm trying to read the MSBuildLocator source and getting quite confused. Is `Microsoft.VisualStudio.Setup.Configuration.Interop` actually needed, along with the `FEATURE_VISUALSTUDIOSETUP` symbol? If we didn't have this `@(Reference)` what would happen? If we didn't define `FEATURE_VISUALSTUDIOSETUP` what would happen?

Reply:
It doesn't appear to build if I remove either the `FEATURE_VISUALSTUDIOSETUP` symbol or the `Microsoft.VisualStudio.Setup.Configuration.Interop` reference.

I am not sure what `FEATURE_VISUALSTUDIOSETUP` is, by googling I only find it in the `msbuild` source: https://github.com/Microsoft/msbuild/blob/master/src/Directory.BeforeCommon.targets#L104

The reference being "development-time" only, this `EmbedInteropTypes` metadata must be what makes the assembly work without the reference: https://github.com/xamarin/xamarin-android/pull/1603/files#diff-5f202698c483a3dc92183954eb4428bfR37